### DISCUSSION TITLE
spatial plot of bias

### DISCUSSION THREAD

**Discussion by timhoar** [NCAR/DART#326] [General]
**Ed note: This issue was originally reported 7 May 2015 and is being manually ported over to GitHub.**

> I am trying to make spatial diagnostics map from which we can see the spatial diatribution of MBE
> (mean bias error) of brightness temperature for certain period. I want to do it using  _obs_diag_output.nc_ .
> So far I have used the matlab code  plot_evolution.m to calclate time series of the RMSE for certain range of the region.
> 
> Can I also draw the spatial map of MBE for the simulation period?
> Then, which matlab diagnostics code do you recommend me to use for my purpose?

Since we have observations and the expected values of the observations, computing the bias is easy. Under certain circumstances, the tricky part is making them available as a spatial plot. Since the observation network is treated as 'randomly located' (even if it is not), the simplest thing to do is to convert the observation sequence files to netCDF and plot them as a scatterplot.  Given the density of observations and the timescales, this may be sub-optimal.

_obs_diag.f90_ can handle an arbitrary number of regions, but the input is tedious and Ryan Torn indicated years ago that it is horribly slow for **lots** of regions. I have not tried to deduce why, although _obs_diag_ has been revised several times since then. _obs_diag.f90_ calculates the mean bias over regions and times ... but was never intended to work on 'gridcell' sized areas - which is what I think is the desire here.

If you are performing a perfect model experiment, you have the True_State.nc and Prior_Diag.nc to work with. That would work
for any model. The question was specifically about CLM, which has a state-space representation of the gridcell constitutent pieces which must be reconstituted into gridcell values. There is a matlab script _clm/matlab/clm_get_var.m_ that will create 'gridcell-based' quantities from CLM and DART restart files. There is also a _clm_plot_var.m_ function.

**Comment by timhoar** [comment]
This is related to the DART_development Issue #87 ... 'heat map' of rejected/accepted observations.
That issue was opened 3 June 2020

**Comment by hkershaw-brown** [comment]
link to the DART_development issue: https://github.com/NCAR/DART_development/issues/87
